// hexo-math/lib/consts.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var INLINE_MATH_REGEX = exports.INLINE_MATH_REGEX = /\$.*?\$/;
var BLOCK_MATH_REGEX = exports.BLOCK_MATH_REGEX = /^\s*\$\$[\s\S]*\$\$\s*$/m;
var INJECTION_REGEX = exports.INJECTION_REGEX = /([\s\S]*)(<\/body>[\s\S]*)/i;
var BODY_REGEX = exports.BODY_REGEX = /<\/body>/i;

var MATH_MARKER = exports.MATH_MARKER = "<!-- Has MathJax -->";

var TEMPLATE = exports.TEMPLATE = "\n<!-- Begin: Injected MathJax -->\n<script type=\"text/x-mathjax-config\">\n  MathJax.Hub.Config(<%= JSON.stringify(config) %>);\n</script>\n\n<script type=\"text/x-mathjax-config\">\n    MathJax.Hub.Queue(function() {\n        var all = MathJax.Hub.getAllJax(), i;\n        for(i=0; i < all.length; i += 1) {\n            all[i].SourceElement().parentNode.className += ' has-jax';\n        }\n    });\n</script>\n\n<script type=\"text/javascript\" src=\"<%= src %>\">\n</script>\n<!-- End: Injected MathJax -->\n";

var DEFAULT_OPTS = exports.DEFAULT_OPTS = {
  src: "//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML",
  config: {
    tex2jax: {
      inlineMath: [['$', '$'], ["\\(", "\\)"]],
      skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
      processEscapes: true
    },
    TeX: {
      equationNumbers: {
        autoNumber: "AMS"
      }
    }
  }
};
