// hexo-math/lib/main.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _math = require("./tag/math");

var _math2 = _interopRequireDefault(_math);

var _inject = require("./filter/inject");

var _inject2 = _interopRequireDefault(_inject);

var _script = require("./script");

var _script2 = _interopRequireDefault(_script);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var MathJax = function () {
  function MathJax(hexo) {
    _classCallCheck(this, MathJax);

    this.hexo = hexo;
    this.tag = new _math2.default(hexo), this.script = new _script2.default(hexo), this.injector = new _inject2.default(hexo, this.script);
  }

  _createClass(MathJax, [{
    key: "register",
    value: function register() {
      var tag = this.tag;
      var injector = this.injector;

      tag.register();
      injector.register();
    }
  }]);

  return MathJax;
}();

exports.default = MathJax;
module.exports = exports['default'];
